azurerm_container_app_environment segfaults when response values are missing in the returned payload

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.14.8

### AzureRM Provider Version

4.69.0

### Affected Resource(s)/Data Source(s)

azurerm_container_app_environment

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "main" {
  name     = "containerapps-rg"
  location = "East US"
}

resource "azurerm_container_app_environment" "apps" {
  name                = "apps-env"
  resource_group_name = azurerm_resource_group.main.name
  location            = azurerm_resource_group.main.location
}

resource "azurerm_container_app" "api" {
  name                         = "api-app"
  resource_group_name          = azurerm_resource_group.main.name
  container_app_environment_id = azurerm_container_app_environment.apps.id
  revision_mode                = "Single"

  ingress {
    target_port      = 8080
    external_enabled = true
    transport        = "http"

    traffic_weight {
      percentage      = 100
      latest_revision = true
    }
  }

  template {
    container {
      name   = "api"
      image  = "mcr.microsoft.com/azuredocs/containerapps-helloworld:latest"
      cpu    = 0.25
      memory = "0.5Gi"
    }
  }
}

resource "azurerm_container_app" "web" {
  name                         = "web-app"
  resource_group_name          = azurerm_resource_group.main.name
  container_app_environment_id = azurerm_container_app_environment.apps.id
  revision_mode                = "Multiple"

  ingress {
    target_port      = 80
    external_enabled = true
    transport        = "http"

    traffic_weight {
      percentage      = 100
      latest_revision = true
    }
  }

  template {
    min_replicas = 1
    max_replicas = 3

    container {
      name   = "web"
      image  = "mcr.microsoft.com/azuredocs/containerapps-helloworld:latest"
      cpu    = 0.5
      memory = "1Gi"

      env {
        name  = "API_URL"
        value = "https://api-app.hashed.eastus.containerApps.k4apps.io"
      }
    }
  }
}

resource "azurerm_container_app_job" "batch" {
  name                         = "batch-job"
  resource_group_name          = azurerm_resource_group.main.name
  container_app_environment_id = azurerm_container_app_environment.apps.id
  location                     = azurerm_resource_group.main.location

  replica_timeout_in_seconds = 600
  replica_retry_limit        = 3

  manual_trigger_config {
    parallelism = 1
    replica_completion_count = 1
  }

  template {
    container {
      name   = "batch"
      image  = "mcr.microsoft.com/samples/containerapps-batchprocessor:latest"
      cpu    = 1.0
      memory = "2Gi"
    }
  }
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v4.69.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x582ef25]

goroutine 126 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containerapps.(*ContainerAppEnvironmentResource).Read.ContainerAppEnvironmentResource.Read.func1({0x90f99e0, 0xc001494930}, {0xc0027b6908, {0x90fd410, 0xc0000110e0}, 0xc001a38d80, 0x0, {0x90fd660, 0xe731520}})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containerapps/container_app_environment_resource.go:419 +0x925
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func2({0x90f99e0, 0xc001494930}, 0xc002a80008?, {0x7682a20?, 0xc0027b6908?})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:67 +0x1ca
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.(*ResourceWrapper).diagnosticsWrapper.diagnosticsWrapper.func13({0x90f99e0?, 0xc001494930?}, 0x1a3185c5000?, {0x7682a20?, 0xc0027b6908?})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:222 +0x59
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0004d0900, {0x90f9938, 0xc001ea8810}, 0xc001a38d80, {0x7682a20, 0xc0027b6908})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:849 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0004d0900, {0x90f9938, 0xc001ea8810}, 0xc001d79380, 0xc001a38c60, {0x7682a20, 0xc0027b6908})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:980 +0xb33
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0001b7d70, {0x90f9938?, 0xc001ea8750?}, 0xc001d1fdb0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/grpc_provider.go:1499 +0x1089
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc00164ce10, {0x90f9938?, 0xc001ea8480?}, 0xc001d1fdb0)
	github.com/hashicorp/terraform-plugin-mux@v0.21.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001e292c0, {0x90f9938?, 0xc001d7fc20?}, 0xc001ea6080)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/tf5server/server.go:944 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x86a2ca0, 0xc001e292c0}, {0x90f9938, 0xc001d7fc20}, 0xc001ea6000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:789 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005b8248, {0x90f9938, 0xc001d7fb90}, 0xc001d3a680, 0xc0019918c0, 0xe6fce68, 0x0)
	google.golang.org/grpc@v1.79.3/server.go:1429 +0x1192
google.golang.org/grpc.(*Server).handleStream(0xc0005b8248, {0x90fda18, 0xc001678000}, 0xc001d3a680)
	google.golang.org/grpc@v1.79.3/server.go:1855 +0xbc6
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.79.3/server.go:1064 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
	google.golang.org/grpc@v1.79.3/server.go:1075 +0x11d

Error: The terraform-provider-azurerm_v4.69.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

I expect the ressource to be created without any errors.

### Actual Behaviour

I get a segfault because it seems `customDomainConfiguration.customDomainVerificationId` and `peerAuthentication.mtls.enabled` are expected. 

### Steps to Reproduce

This has to be done by reproducing the response expected by the above mentioned Terraform script to see the error. I am working on a local Azure emulator https://miniblue.io/ that is how I found out about this issue

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/containerapps/container_app_environment_resource.go
+++ b/internal/services/containerapps/container_app_environment_resource.go
@@ -409,14 +409,20 @@ func (r ContainerAppEnvironmentResource) Read() sdk.ResourceFunc {
 						}
 					}
 
-					state.CustomDomainVerificationId = pointer.From(props.CustomDomainConfiguration.CustomDomainVerificationId)
 					state.PublicNetworkAccess = pointer.FromEnum(props.PublicNetworkAccess)
 					state.ZoneRedundant = pointer.From(props.ZoneRedundant)
 					state.StaticIP = pointer.From(props.StaticIP)
 					state.DefaultDomain = pointer.From(props.DefaultDomain)
 					state.WorkloadProfiles = helpers.FlattenWorkloadProfiles(props.WorkloadProfiles)
 					state.InfrastructureResourceGroup = pointer.From(props.InfrastructureResourceGroup)
-					state.Mtls = pointer.From(props.PeerAuthentication.Mtls.Enabled)
+
+					if props.CustomDomainConfiguration != nil {
+						state.CustomDomainVerificationId = pointer.From(props.CustomDomainConfiguration.CustomDomainVerificationId)
+					}
+
+					if props.PeerAuthentication != nil && props.PeerAuthentication.Mtls != nil {
+						state.Mtls = pointer.From(props.PeerAuthentication.Mtls.Enabled)
+					}
 				}
 			}
 

